src/Microsoft.OData.Core/UriParser/Binders/SelectPathSegmentTokenBinder.cs

```diff
@@ -178,18 +178,24 @@ internal static bool TryBindAsOperation(PathSegmentToken pathToken, IEdmModel mo
             }
 
             // Only filter if there is more than one and its needed.
-            if (possibleFunctions.Count() > 1)
+            if(possibleFunctions.HasAny())

```

Comment:
Not sure whether the cost of doing a HasAny() and then calling Count() if there are any is worth the potential savings in shortcutting the next set of logic if there aren't any.

This whole thing should be rewritten to avoid the multiple iterations. For now, I'd skip doing the initial HasAny().

Reply:
Honestly, calling Count() to start with is better from a performance standpoint:
1. If IEnumerable is a ICollection Count() will cast and invoke the property **Count** on the collection which is relatively cheap.
2. Even if IEnumerable is not a ICollection but the collection is empty the underlying O(n) algorithm has the same cost as invoking HasAny() for an empty collection. If the collection is not empty, then we need to call Count() anyway.